test/powershell/Language/Scripting/NativeExecution/NativeCommandProcessor.Tests.ps1

```diff
@@ -347,3 +347,113 @@ Describe "Run native command from a mounted FAT-format VHD" -tags @("Feature", "
         $result | Should -BeExactly $expected
     }
 }
+
+Describe "Scriptblock passed as arguments to native executables" -tag @("CI") {
+    BeforeAll {
+        $skipTest = $false
+        if (-not (Get-Command PowerShell)) {
+            $skipTest = $true
+            $reason = "PowerShell is not installed"
+        }
+
+        if (-not (Get-ExperimentalFeature PSNativeScriptBlockArgument).Enabled) {
+            $skipTest = $true
+            $reason = "Experimental feature PSNativeScriptBlockArgument is not enabled"
+        }
+
+        $pwsh = Join-Path -Path $PSHOME -ChildPath pwsh
+    }
+
+    # current behavior hasn't changed for PowerShell
+    Context "Scriptblocks are encoded when calling PowerShell or pwsh" {
+        It  "A direct call to the pwsh symlink with 'pwsh -command {get-date}' is encoded correctly" {
+            if ($skipTest) {
+                Set-ItResult -skipped -Because $reason
+            }
+            pwsh -command {get-date} | Should -BeOfType DateTime
+        }
+
+        It  "A direct call to 'pwsh -command {get-date}' is encoded correctly" {
+            if ($skipTest) {
+                Set-ItResult -skipped -Because $reason
+            }
+            & $pwsh -command {get-date} | Should -BeOfType DateTime
+        }
+
+        It "A direct call to 'powershell -command {get-date}' is encoded correctly" {
+            if ($skipTest) {
+                Set-ItResult -skipped -Because $reason
+            }
+            powershell -command {get-date} | Should -BeOfType DateTime
+        }
+    }
+
+    Context "ScriptBlocks are not encoded when calling applications other than PowerShell or pwsh" {
+        It "scriptblock will be passed as is." {
+            if ($skipTest) {
+                Set-ItResult -skipped -Because $reason
+            }
+            $result = testexe -echoargs a b {get-date}
+            $result.Count | Should -Be 3
+            $result[0] | Should -BeExactly 'Arg 0 is <a>'
+            $result[1] | Should -BeExactly 'Arg 1 is <b>'
+            $result[2] | Should -BeExactly 'Arg 2 is <{get-date}>'
+        }
+
+        It "a script block with a space will be passed as is" {
+            if ($skipTest) {
+                Set-ItResult -skipped -Because $reason
+            }
+            $result = testexe -echoargs a b { get-date }
+            $result.Count | Should -Be 3
+            $result[0] | Should -BeExactly 'Arg 0 is <a>'
+            $result[1] | Should -BeExactly 'Arg 1 is <b>'
+            $result[2] | Should -BeExactly 'Arg 2 is <{ get-date }>'
+        }
+
+        It "a script block with a space and new line will be passed as is" {
+            if ($skipTest) {
+                Set-ItResult -skipped -Because $reason
+            }
+            $result = testexe -echoargs a b { get-date
+            get-location }
```

Comment:
If you don't have whitespace in front of this line here, would that be treated as one arg or two?